library/entropy.c

```diff
@@ -466,28 +466,27 @@ int mbedtls_entropy_update_nv_seed( mbedtls_entropy_context *ctx )
 #if defined(MBEDTLS_FS_IO)
 int mbedtls_entropy_write_seed_file( mbedtls_entropy_context *ctx, const char *path )
 {
-    int ret = MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR;
-    FILE *f;
+    int ret;
+    FILE *f = NULL;
     unsigned char buf[MBEDTLS_ENTROPY_BLOCK_SIZE];
 
-    if( ( f = fopen( path, "wb" ) ) == NULL )
-        return( MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR );
-
     if( ( ret = mbedtls_entropy_func( ctx, buf, MBEDTLS_ENTROPY_BLOCK_SIZE ) ) != 0 )
         goto exit;
 
-    if( fwrite( buf, 1, MBEDTLS_ENTROPY_BLOCK_SIZE, f ) != MBEDTLS_ENTROPY_BLOCK_SIZE )
+    ret = MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR;
```

Comment:
This is a very clever way of organising the flow so that we only need to assign the `MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR` once. But I am afraid that this is comes at a cost of readability, could we please use the basic
```c
ret = MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR;
goto exit;
```
exception handling, even if it results in duplication?

Reply:
Agree; will rework. I had pushed the changes before I fully examined the overall structure. Will try to adhere to the existing style in future. Thanks!